src/menu/Menu.ts

```diff
@@ -0,0 +1,311 @@
+
+		if (label) {
+			const properties = typeof label === 'string' ? { label } : label;
+			return w(MenuItem, assign({
+				disabled,
+				getAriaProperties: this._getLabelAriaProperties.bind(this, id),
+				hasMenu: true,
+				overrideClasses: overrideClasses || css,
+				onClick: this.onLabelClick,
+				onKeypress: this.onLabelKeypress
+			}, properties));
+		}
+	}
+
+	protected _afterCreate(element: HTMLElement) {
+		const { animate = true } = this.properties;
+		this._initialRender = false;
+
+		if (animate) {
+			const hidden = element.classList.contains(css.hidden);
+			this._wasOpen = !hidden;
+
+			if (hidden) {
+				element.style.height = '0';
+			}
+		}
+	}
+
+	protected _afterUpdate(element: HTMLElement) {
+		const { animate = true, hidden = this._getDefaultHidden(), label } = this.properties;
+
+		if (!label) {
+			return;
+		}
+
+		if (!animate) {
+			// In case `animate` was previously `true`, remove any `height` property set on the node.
+			element.style.height = null;
+			return;
+		}
+
+		if (typeof animate === 'function') {
+			return animate(element, hidden);
+		}
+
+		this._animate(element, hidden);
+	}
+
+	protected _animate(element: HTMLElement, hidden: boolean) {
+		// Assuming the menu has an `auto` height, manually apply the scroll height (or max-height if specified),
+		// and animate to and from that.
+		requestAnimationFrame(() => {
+			const height = getMenuHeight(element);
+			this._wasOpen = !hidden;
+
+			if (hidden) {
+				// Temporarily remove any transition to prevent the element from animating to `height`.
+				const transition = element.style.transition;
+				element.style.transition = null;
+
+				element.style.height = height + 'px';
+				element.style.transition = transition;
+				requestAnimationFrame(() => {
+					element.style.height = '0';
+				});
+			}
+			else {
+				element.style.height = `${height}px`;
+			}
+		});
+	}
+
+	protected _getDefaultHidden() {
+		const { disabled, label } = this.properties;
+
+		if (label && disabled) {
+			return true;
+		}
+
+		return label ? true : false;
+	}
+
+	protected _getLabelAriaProperties(id: string): { [key: string]: string; } {
+		const { hidden = this._getDefaultHidden() } = this.properties;
+		return {
+			'aria-controls': id,
+			'aria-expanded': String(!hidden)
+		};
+	}
+
+	protected _getMenuClasses(hidden: boolean, isSubMenu: boolean) {
+		const { animate = true, nested } = this.properties;
+		const classes = [ css.menu ];
+
+		if (this._initialRender || !animate || !isSubMenu) {
+			classes.push(hidden ? css.hidden : css.visible);
+		}
+
+		if (nested) {
+			classes.push(css.nestedMenu);
+		}
+
+		if (isSubMenu) {
+			classes.push(css.subMenu);
+		}
+
+		return classes;
+	}
+
+	protected _toggleDisplay(isHidden?: boolean) {
+		const {
+			onRequestHide,
+			onRequestShow
+		} = this.properties;
+
+		if (typeof isHidden === 'undefined') {
+			const { hidden = this._getDefaultHidden() } = this.properties;
```

Comment:
`isHidden` is a way to force which state is requested. I'll update the parameter name to make that obvious.